File: flutter/lib/data/dao/flashcard_dao.dart
import 'package:flashcard/data/entities/flashcard_db_entity.dart';
import 'package:floor/floor.dart';

@dao
abstract class FlashcardDao {
  @insert
  Future<void> createFlashcard(FlashcardDbEntity flashcard);

  @Query('SELECT * FROM flashcard WHERE deckId = :deckId')
  Future<List<FlashcardDbEntity>> getAllFlashcardsFromDeckId(int deckId);

  @update
  Future<void> updateFlashcard(FlashcardDbEntity flashcard);

  @delete
  Future<void> deleteFlashcard(FlashcardDbEntity flashcard);

  @Query('DELETE FROM flashcard WHERE id = :flashcardId')
  Future<void> deleteFlashcardById(int flashcardId);
}

File: android/app/src/main/java/com/synergyboat/flashcardAi/data/dao/FlashcardDao.kt
package com.synergyboat.flashcardAi.data.dao


import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import com.synergyboat.flashcardAi.data.entities.FlashcardEntity

@Dao
interface FlashcardDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun createFlashcard(flashcard: FlashcardEntity): Long

    @Query("SELECT * FROM flashcard WHERE deckId = :deckId")
    suspend fun getAllFlashcardsFromDeckId(deckId: Long): List<FlashcardEntity>

    @Update
    suspend fun updateFlashcard(flashcard: FlashcardEntity)

    @Delete
    suspend fun deleteFlashcard(flashcard: FlashcardEntity)

    @Query("DELETE FROM flashcard WHERE id = :flashcardId")
    suspend fun deleteFlashcardById(flashcardId: Long)
}